spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -151,5 +158,18 @@ void configure(String channelName, MessageChannel channel, ProducerProperties pr
 				T extendedProducerProperties);
 
 	}
+	
+	/**
+	 * Will add any globally available interceptors to the MessageChannel
+	 * 
+	 * See: https://github.com/spring-cloud/spring-cloud-stream/issues/1219
+	 */
+	private void instrumentChannel(MessageChannel messageChannel) {
+		Collection<GlobalChannelInterceptorWrapper> interceptors =
+				this.beanFactory.getBeansOfType(GlobalChannelInterceptorWrapper.class).values();
+		for (GlobalChannelInterceptorWrapper globalChannelInterceptorWrapper : interceptors) {
+			((AbstractMessageChannel)messageChannel).addInterceptor(globalChannelInterceptorWrapper.getChannelInterceptor());
```

Comment:
You do this unconditionally. Any `@GlobalChannelInterceptor` contributes `patterns` to the `GlobalChannelInterceptorWrapper`. So, if we go this way you have to use its `getPatterns()` over the `PatternMatchUtils.smartMatch()` to be sure that we definitely need it apply such an interceptor to the channel.

The `(AbstractMessageChannel)` cast is not good as well since it is a class and in case of Proxy the cast are not going to work. You need to cast to the `ChannelInterceptorAware` (if that is possible at all)...

See `GlobalChannelInterceptorProcessor` for more information.

Moreover I'm still not sure why you didn't go the way to inject all the end-user `MessageChannelConfigurer`s, which are going to be called from the `SubscribableChannelBindingTargetFactory` for any channel created by the SCSt.

Also pay attention how you call `this.beanFactory.getBeansOfType()` for each channel we create here.

Does it make sense to you?

Reply:
The reason why it is not in `SubscribableChannelBindingTargetFactory` is because that is where we add our interceptors, the Tracing and others are added via `GlobalChannelInterceptorProcessor` so if I add this code there I need to make sure that the interceptors added via `GlobalChannelInterceptorProcessor` aren't added again. In other words it is only the concern for dynamic channels. 
The rest makes sense, so I'll polish.
Basically with everything that has been discussed here about `TraceChannelInterceptor` created a bit of a confusion . . . the solution turned out to be much simpler -  he interceptor was simply never added to a channel.
Now I can see all the sleuth traces